lib/cisco_node_utils/itd_device_group.rb

```diff
@@ -0,0 +1,210 @@
+# February 2016, Sai Chintalapudi
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for itd_device_group
+  class ItdDeviceGroup < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name.length > 0
+      @name = name
+
+      set_args_keys_default
+      create if instantiate
+    end
+
+    def self.itds
+      hash = {}
+      group_list = []
+      groups = config_get('itd_device_group',
+                          'all_itd_device_groups')
+      return hash if groups.nil?
+
+      groups.each do |group|
+        # The show cmd shows more than name if the probe is not confgured yet
+        # itd device-group abc frequency 10 timeout 5 retry-down-count 3 ...
+        # so filter it out to just get the name
```

Comment:
You should not need to filter if you change the `get_value` yaml entry to the following.

`get_value: '/^itd device-group (\S+)/'`